File: src/gallery/image.cpp
#include "image.h"

Image::Image()
{
    id = "";
    parentId = "";
    name = "";
    author = "";
    date = "";
    size = "";
    description = "";
}

Image::Image(QString id, QString parentId)
{
    this->id = id;
    this->parentId = parentId;
    name = "";
    author = "";
    date = "";
    size = "";
    description = "";
}

void Image::write(QJsonObject &jsonObj) const
{
    jsonObj["id"] = id;
    jsonObj["parentId"] = parentId;
}

void Image::read(const QJsonObject &jsonObj)
{
    id = jsonObj["id"].toString();
    parentId = jsonObj["parentId"].toString();
}

QString Image::getId() const
{
    return id;
}

void Image::setId(const QString &value)
{
    id = value;
}

QString Image::getParentId() const
{
    return parentId;
}

void Image::setParentId(const QString &value)
{
    parentId = value;
}

QString Image::getName() const
{
    return name;
}

void Image::setName(const QString &value)
{
    name = value;
}

QString Image::getAuthor() const
{
    return author;
}

void Image::setAuthor(const QString &value)
{
    author = value;
}

QString Image::getDate() const
{
    return date;
}

void Image::setDate(const QString &value)
{
    date = value;
}

QString Image::getSize() const
{
    return size;
}

void Image::setSize(const QString &value)
{
    size = value;
}

QString Image::getDescription() const
{
    return description;
}

void Image::setDescription(const QString &value)
{
    description = value;
}

File: src/ce/itcr/invincible/data/Image.java
package ce.itcr.invincible.data;

public class Image {

    private int id;
    private String parentId;
    private String name;
    private String author;
    private int year;
    private int size;
    private String description;
    private String metaData;
    private String compressedData;

    public Image() { }

    public Image(int id, String parentId, String name, String author, int year, int size, String description, String metaData, String compressedData) {
        this.id = id;
        this.parentId = parentId;
        this.name = name;
        this.author = author;
        this.year = year;
        this.size = size;
        this.description = description;
        this.metaData = metaData;
        this.compressedData = compressedData;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getParentId() {
        return parentId;
    }

    public void setParentId(String parentId) {
        this.parentId = parentId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getMetaData() {
        return metaData;
    }

    public void setMetaData(String metaData) {
        this.metaData = metaData;
    }

    public String getCompressedData() {
        return compressedData;
    }

    public void setCompressedData(String compressedData) {
        this.compressedData = compressedData;
    }
}
